core/trino-main/src/main/java/io/trino/operator/HashSemiJoinOperator.java

```diff
@@ -160,7 +160,7 @@ public SemiJoinPages(SetSupplier channelSetFuture, int probeJoinChannel, Optiona
 
             this.channelSetFuture = requireNonNull(channelSetFuture, "channelSetFuture is null").getChannelSet();
             this.probeJoinChannel = probeJoinChannel;
-            this.probeHashChannel = requireNonNull(probeHashChannel, "probeHashChannel is null");
+            this.probeHashChannel = requireNonNull(probeHashChannel, "probeHashChannel is null").orElse(-1);
```

Comment:
Make `-1` a constant named somethiing like `NO_PRECOMPUTED_HASH_CHANNEL`

Reply:
Sure, but it's already a pretty common idiom to use `-1` as a sentinel for "not a valid channel". See [PartitionedOutputOperator](https://github.com/trinodb/trino/blob/6bbe9f1092db394c45a005040c1a03af7e765520/core/trino-main/src/main/java/io/trino/operator/PartitionedOutputOperator.java#L473) via [LocalExecutionPlanner](https://github.com/trinodb/trino/blob/6bbe9f1092db394c45a005040c1a03af7e765520/core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java#L451) or [InMemoryHashAggregationBuilder](https://github.com/trinodb/trino/blob/6bbe9f1092db394c45a005040c1a03af7e765520/core/trino-main/src/main/java/io/trino/operator/aggregation/builder/InMemoryHashAggregationBuilder.java#L353)